lib/bolt/transport/ssh/exec_connection.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+require 'open3'
+
+module Bolt
+  module Transport
+    class SSH < Simple
+      class ExecConnection
+        attr_reader :user, :target
+
+        def initialize(target)
+          @target = target
+          ssh_config = Net::SSH::Config.for(target.host)
+          @user = @target.user || ssh_config[:user] || Etc.getlogin
```

Comment:
I think this actually needs to be set as we use it elsewhere in the codebase, like setting the owner for tmpdir https://github.com/puppetlabs/bolt/blob/master/lib/bolt/shell/bash/tmpdir.rb#L9 or setting error messages https://github.com/puppetlabs/bolt/blob/master/lib/bolt/shell/bash.rb

Reply:
Ugh that's unfortunate. I wish we could use `ssh -G` to parse the config (rather than relying on `Net::SSH`) but it appears that's only available in newer SSH versions. :(

Reply:
Yeah :(